mypy/checkexpr.py

```diff
@@ -2341,7 +2341,9 @@ def apply_inferred_arguments(
         for i, inferred_type in enumerate(inferred_args):
```

Comment:
s/enumerate/zip? (ideally `zip(strict=True)` but we still need to run on 3.9, unfortunately)